draco/analysis/flagging.py

```diff
@@ -416,6 +416,42 @@ def process(self, data):
         return out
 
 
+class BadGainMask(task.SingleTask):
+    """Get a mask of regions with bad gain.
+
+    Assumes that bad gains are set to 1.
+    """
+
+    def process(self, data):
+        """Generate a time-freq mask.
+
+        Parameters
+        ----------
+        data : :class:`andata.CorrData` or :class:`containers.TimeStream` object
+            Data containing the weights to be flagged. Must have a `gain` dataset.
+
+        Returns
+        -------
+        mask : RFIMask container
+            Time-freq mask
+        """
+
+        # Ensure data is distributed in frequency
+        data.redistribute("freq")
+
+        # Boolean mask where gains are bad across all baselines.
+        mask = np.all(data.gain[:] == 1.0, axis=1).allgather()
```

Comment:
This kind of floating-point comparison makes me nervous, but I can't find a simple `np.allclose()` analog for `np.all(..., axis=1)`, so I guess it's fine?

Reply:
I've changed this check in a way that should work